NDTensors/src/lib/SetParameters/src/Base/abstractarray.jl

```diff
@@ -0,0 +1,7 @@
+function set_eltype(type::Type{<:AbstractArray}, elt::Type)
+  return set_parameter(type, Position(1), elt)
+end
+
+function set_ndims(type::Type{<:AbstractArray}, ndim::Int)
+  return set_parameter(type, Position(2), ndim)
+end
```

Comment:
I think these are reasonable defaults, though we need to remember that if it is a wrapped array type then the wrapped array also will need to be adjusted. So we will need specialized definitions for wrapped array types like `SubArray`.

Reply:
One way to handle wrapper types like `SubArray` could be to also introduce a function `set_parenttype`. Then we could have definitions like:
```julia
function set_eltype(type::Type{<:SubArray}, elt::Type)
  new_parenttype = set_eltype(parenttype(type), elt)
  return set_parenttype(set_parameter(type, Position(1), elt), new_parenttype)
end
```
which would work generically for many (maybe all) wrapped array types.

That probably requires making `Unwrap` a dependency of `SetParameters`, since that is where `parenttype` is defined and we don't want to repeat that definition here. In the end, it might make sense to move `parenttype` from `Unwrap` to `SetParameters` but to keep this PR simpler let's leave that for future work.

Probably the best/most general design of `set_eltype` would be to make it a trait function based around the trait `IsWrappedArray` from `Unwrap`, which then has one simpler definition of `set_eltype` when `!IsWrappedArray` (the current definition in this PR), and another when `IsWrappedArray` which is the definition using `parenttype` that I wrote above.

Reply:
@mtfishman Thank you for these ideas! I am still working/thinking about the best way to implement `set_eltype` and `set_ndims`. and I will work to include these suggestions today

Reply:
@mtfishman  I have made some changes to the `set_eltype` and `set_ndims` system and mostly have deleted the functions from NDTensors. I haven't yet made a `Wrapped` based type system though I do plan to.  Right now I have simply created multiple `set_x` functions based on the type input. I am pushing my changes so I can test on CUDA (so far I have only tested using `Metal` and CPU) and will continue to work on this PR. Thanks!

Reply:
Sounds good, makes sense as an initial step.